src/controls/IRoomControllerV2.ts

```diff
@@ -0,0 +1,380 @@
+            control.states,
+            'overrideReason',
+            'number',
+            'value',
+            {
+                states: {
+                    0: 'None',
+                    1: 'Presence -> Comfort',
+                    2: 'Window open -> Eco+',
+                    3: 'Comfort',
+                    4: 'Eco',
+                    5: 'Eco+',
+                    6: 'Heat Up',
+                    7: 'Cool Down',
+                },
+            },
+        );
+        await this.createSimpleControlStateObjectAsync(
+            control.name,
+            uuid,
+            control.states,
+            'prepareState',
+            'number',
+            'value',
+            {
+                states: {
+                    '-1': 'Cooling Down',
+                    0: 'None',
+                    1: 'Heating Up',
+                },
+            },
+        );
+        await this.createBooleanControlStateObjectAsync(control.name, uuid, control.states, 'openWindow', 'indicator');
+        await this.createBooleanControlStateObjectAsync(control.name, uuid, control.states, 'useOutdoor', 'indicator');
+
+        await this.createSimpleControlStateObjectAsync(
+            control.name,
+            uuid,
+            control.states,
+            'frostProtectTemperature',
+            'number',
+            'value.temperature',
+        );
+        await this.createSimpleControlStateObjectAsync(
+            control.name,
+            uuid,
+            control.states,
+            'heatProtectTemperature',
+            'number',
+            'value.temperature',
+        );
+        // Add event handler to frost/heat protection values so we re-calc target min/max if they change
+        this.addStateEventHandler(uuid + '.frostProtectTemperature', async () => {
+            await this.updateTempTargetMinMax();
+        });
+        this.addStateEventHandler(uuid + '.heatProtectTemperature', async () => {
+            await this.updateTempTargetMinMax();
+        });
+
+        await this.updateStateObjectAsync(
+            uuid + '.activeMode',
+            {
+                name: control.name + ': activeMode',
+                read: true,
+                write: true,
+                type: 'number',
+                role: 'level',
+                states: {
+                    0: 'Economy',
+                    1: 'Comfort',
+                    2: 'Fabric Protection',
+                    3: 'Manual',
+                },
+            },
+            control.states.activeMode,
+            async (name: string, value: any) => {
+                await this.setStateAck(name, value);
+                await this.updateTempTargetMinMax();
+            },
+        );
+        this.addStateChangeListener(uuid + '.activeMode', (oldValue: OldStateValue, newValue: CurrentStateValue) => {
+            this.sendCommand(control.uuidAction, 'override/' + newValue);
+        });
+
+        await this.updateStateObjectAsync(
```

Comment:
I think it would be good to create a method that creates a temperature object, it can be reused everywhere where you have `updateStateObjectAsync()` for a temperature. I would probably place this method in this class as it is only used for IRC.

Reply:
Sorry, but I see very little value in adding that.